card.py

```diff
@@ -48,3 +48,16 @@ def deserialize(cls, data: bytes):
         assert_deserialization_finished(buf)
         return cls(name, creator, image, riddle, None)
 
+    def encrypt(self):
+        """Encypt the card's image using its solution."""
+        self.image.encrypt(self.solution.encode())
```

Comment:
I'm not really sure what you mean? `encrypt` and `solve` are error-agnostic. If an error will occur it will be caught by the `try-except` block in `main`. I also don't see a reason why `str.encode()` or `CryptImage.encrypt()` will throw an exception.